src/AutoRest.CSharp/Mgmt/Output/Samples/MgmtSampleProvider.cs

```diff
@@ -430,9 +430,18 @@ private MethodBodyStatement BuildGetCollectionStatement(ResourceCollection colle
             var statements = new List<MethodBodyStatement>();
 
             var resourceName = collection.Resource.ResourceName;
-            statements.Add(
-                BuildGetResourceStatement(parent, example, client, out var parentVar)
-                );
+            TypedValueExpression? parentVar;
+            // get parent resource
+            if (parent is MgmtExtension extension && extension.ArmCoreType == typeof(ArmResource))
```

Comment:
I think we should put a big if...else branch to differentiate this case and other cases, instead of checking `parentVar == null`, like:
```csharp
if (parent is MgmtExtension extension && extension.ArmCoreType == typeof(ArmResource))
{
    writeGetCollection();
    writeScopedResourceId();
    writeExtraConstructorParameters();
    writeCallMethodToGetCollection();
}
else
{
    writeGetResource();
    writeGetCollection();
    writeExtraConstructorParameters();
    statements.Add(
        Declare(collection.Type, "collection", parentVar.Invoke(getResourceCollectionMethodName, arguments), out instance)
    );
}
```

Reply:
let me try if this could work

Reply:
Fixed!